packages/actor-rdf-source-identifier-file-content-type/lib/ActorRdfSourceIdentifierFileContentType.ts

```diff
@@ -0,0 +1,44 @@
+import {IActionHttp, IActorHttpOutput} from "@comunica/bus-http";
+import {ActorRdfSourceIdentifier, IActionRdfSourceIdentifier,
+  IActorRdfSourceIdentifierOutput} from "@comunica/bus-rdf-source-identifier";
+import {Actor, IActorArgs, IActorTest, Mediator} from "@comunica/core";
+
+/**
+ * A comunica File Content Type RDF Source Identifier Actor.
+ */
+export class ActorRdfSourceIdentifierFileContentType extends ActorRdfSourceIdentifier {
+
+  public readonly allowedMediaTypes: string[];
+  public readonly mediatorHttp: Mediator<Actor<IActionHttp, IActorTest, IActorHttpOutput>,
+    IActionHttp, IActorTest, IActorHttpOutput>;
+
+  constructor(args: IActorRdfSourceIdentifierFileContentTypeArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfSourceIdentifier): Promise<IActorTest> {
+    if (!action.sourceValue.startsWith('http')) {
+      throw new Error(`Actor ${this.name} can only detect files hosted via HTTP(S).`);
+    }
+    const headers: Headers = new Headers();
+    headers.append('Accept', this.allowedMediaTypes.join(';'));
+    const httpAction: IActionHttp = { input: action.sourceValue, init: { headers, method: 'HEAD' } };
+    const httpResponse: IActorHttpOutput = await this.mediatorHttp.mediate(httpAction);
```

Comment:
I thought so as well initially. Most of that code used to be in `run`, but I changed it because of the following reasons:

1. In a way, you could see the real action of this actor being the labelling of the source. Where the `test` is then checking if the actor is able to label the input, and reject otherwise. If all of this logic would be inside the `run`, then we would have to allow `run` to reject as well, which is something that the mediator is not able to handle (by default).
2. It was difficult to come up with a mediator for this. We would have to create a new mediator that `runs` all actors, and only takes the output of the one that does not reject. (Which makes the behaviour of `run` closely resemble the intention of `test`)